### PR TITLE
chore: add more integration tests for email workflows

### PR DESCRIPTION
**Pull request by junminahn**

### CODE REVIEW

File: lambda/app/src/controllers/team.ts [hunk 1]
```diff
@@ -88,7 +88,7 @@ export const deleteTeam = async (user: User, id: string) => {
   );
 
   const team = await models.team.findOne({ where: { id } });
-  sendTemplate(EMAILS.TEAM_DELETED, { team });
+  await sendTemplate(EMAILS.TEAM_DELETED, { team });
```
[the remark this answers is not in the document]
Comment (junminahn) [reply lambda/app/src/controllers/team.ts:91]: this is a good argument, and we might not want to wait for the emails sent out.
The reasons behind this is that:
1. It is more robust to wait for the email function invoked in the regression tests as sometimes it fails to capture the email data.
2. in terms of the conventions, this was the only place not waiting for this func call.

So, I believe, if waiting for email calls reduces the user experience in prod, we can re-consider revisiting them.

File: lambda/app/src/github.ts [hunk 2]
```diff
@@ -61,29 +61,24 @@ export const dispatchRequestWorkflow = async (formData: GitHubRequestDispatchInp
...
+  return Promise.all(
+    issuesRes.data.map((issue) => {
+      return octokit.rest.issues.update({
+        owner: process.env.GH_OWNER,
+        repo: process.env.GH_REPO,
+        issue_number: issue.number,
+        state: 'closed',
+      });
+    }),
+  );
```
[the remark this answers is not in the document]
Comment (junminahn) [reply lambda/app/src/github.ts:83]: we have `try... catch` blocks on the `routes` level, so it is not necessary to have extra error handling n the lower level.

File: lambda/app/src/helpers/token.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
+import { verify, sign } from 'jsonwebtoken';
+import { User } from '@lambda-shared/interfaces';
+
+const VERIFY_USER_SECRET = process.env.VERIFY_USER_SECRET || 'asdf';
```
[the remark this answers is not in the document]
Comment (junminahn) [reply lambda/app/src/helpers/token.ts:4]: this env is listed in `nodemon.example.json` already

File: lambda/app/src/github.ts (hunk 2 above)
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply lambda/app/src/github.ts:83]: sounds good

File: lambda/app/src/helpers/token.ts (hunk 3 above)
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply lambda/app/src/helpers/token.ts:4]: awesome

File: lambda/app/src/controllers/team.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (thegentlemanphysicist) [reply lambda/app/src/controllers/team.ts:91]: awesome, just wanted to understand the reasoning behind it

**Review by thegentlemanphysicist — APPROVED** [review]

**Pull request merged**